CMakeLists.txt

```diff
@@ -231,14 +234,14 @@ add_library(
   TritonPythonBackend::triton-python-backend ALIAS triton-python-backend
 )
 
-target_compile_features(triton-python-backend PRIVATE cxx_std_11)
+target_compile_features(triton-python-backend PRIVATE cxx_std_${TRITON_MIN_CXX_STANDARD})
```

Comment:
Why not use the existing `CMAKE_CXX_STANDARD` variable. For example, `cxx_std_${CMAKE_CXX_STANDARD}`?

Reply:
Seems like some of the libraries follow this pattern: https://github.com/IntelRealSense/librealsense/blob/ea975f8641a83a38a3f2ef3baa4cc42e94a75219/CMake/lrs_macros.cmake#L24

Reply:
we do not explicitly define and set `CMAKE_CXX_STANDARD`, nor do we propagate it.To use `CMAKE_CXX_STANDARD`, we would need to set  it first or see if it was passed by user through arguments, since we want to move to C++17, we will also need to check if passed `CMAKE_CXX_STANDARD` is >=17. Plus, from what I've researched so far, if `CMAKE_CXX_STANDARD` is used, it is typically used through `set(CMAKE_CXX_STANDARD  <STANDARD>)`, which is too restrictive, as I mentioned in the slack discussion:

it will force the project to be compiled in specified standard only, it may cause some troubles with later upgrades of Ubuntu or other upgrades.  target_compile_features just specifies [minimum required standard](https://www.scivision.dev/cmake-force-old-language-standard/). Thus, we are not restricted. Some of related discussions on this topic: https://sarcasm.github.io/notes/dev/cmake.html , https://github.com/AcademySoftwareFoundation/OpenColorIO/issues/1525 and https://github.com/AcademySoftwareFoundation/aswf-sample-project/pull/8#issuecomment-952962520

Reply:
How about keeping everything as-is (was) and just making sure that `CMAKE_CXX_STANDARD` is propagated to all the repos? We can set it once during the build.py script by specifying `-DCMAKE_CXX_STANDARD=11,17,etc`.

Reply:
What if users will build with cmake explicitly? Then we would need to add more code that checks that `CMAKE_CXX_STANDARD` is set, and that the value is >= 17. Plus, propagation would mostly be done through cmake cache, i.e. code will still be similar to what I currently have. 

I don't see any benefit in this, could you please elaborate your concerns?

Reply:
> keeping everything as-is (was)

Could you please clarify this as well? Currently, min required standard is 11, which is hard-coded with the following:
`target_compile_features(triton-python-backend PRIVATE cxx_std_11)`

Reply:
I think now I have a better idea about the challenges involved. thanks for clarifying. Resolving.

Reply:
No problem! Thanks for the discussion.